pkg/cableengine/cableengine.go

```diff
@@ -242,28 +244,28 @@ func (i *engine) InstallCable(endpoint *v1.Endpoint) error {
 	}
 
 	i.Lock()
-	i.natDiscoveryPending[endpoint.Spec.CableName]++
+	i.natDiscoveryPending[endpoint.Spec.GetFamilyCableName(family)]++
 	i.Unlock()
 
-	i.natDiscovery.AddEndpoint(endpoint)
+	i.natDiscovery.AddEndpoint(endpoint, family)
 
 	return nil
 }
 
-func (i *engine) RemoveCable(endpoint *v1.Endpoint) error {
+func (i *engine) RemoveCable(endpoint *v1.Endpoint, family k8snet.IPFamily) error {
 	if endpoint.Spec.ClusterID == i.localCluster.ID {
 		logger.V(log.DEBUG).Infof("Cables are not added/removed for the local cluster, skipping removal")
 		return nil
 	}
 
-	logger.Infof("Removing Endpoint cable %q", endpoint.Spec.CableName)
-
-	i.natDiscovery.RemoveEndpoint(endpoint.Spec.CableName)
+	logger.Infof("Removing Endpoint IP%v cable %q", family, endpoint.Spec.CableName)
```

Comment:
Minor but I think this reads better.

```suggestion
	logger.Infof("Removing Endpoint IPv%v cable %q", family, endpoint.Spec.CableName)
```